src/daft-sql/src/planner.rs

```diff
@@ -246,32 +248,37 @@ impl SQLPlanner {
                     .collect::<SQLPlannerResult<Vec<_>>>()?;
             }
         }
+        let mut projections = Vec::with_capacity(selection.projection.len());
+        let mut projection_fields = Vec::with_capacity(selection.projection.len());
+        for expr in &selection.projection {
+            let exprs = self.select_item_to_expr(expr, &schema)?;
 
-        // split the selection into the groupby expressions and the rest
-        let (groupby_selection, to_select) = selection
-            .projection
-            .iter()
-            .map(|expr| self.select_item_to_expr(expr))
-            .collect::<SQLPlannerResult<Vec<_>>>()?
-            .into_iter()
-            .flatten()
-            .partition::<Vec<_>, _>(|expr| {
-                groupby_exprs
-                    .iter()
-                    .any(|e| expr.input_mapping() == e.input_mapping())
-            });
+            let fields = exprs
+                .iter()
+                .map(|expr| expr.to_field(&schema).map_err(PlannerError::from))
+                .collect::<SQLPlannerResult<Vec<_>>>();
+            let fields = fields?;
```

Comment:
```suggestion
            let fields = exprs
                .iter()
                .map(|expr| expr.to_field(&schema).map_err(PlannerError::from))
                .collect::<SQLPlannerResult<Vec<_>>>()?;
```

Reply:
good catch. I forgot to remove this after debugging!